src/http/conn.rs

```diff
@@ -490,7 +497,7 @@ enum Reading {
 
 enum Writing<B> {
     Init,
-    Body(Encoder, Option<Cursor<B>>),
+    Body(Encoder, Option<Cursor<B>>, BytesMut),
```

Comment:
I tend to try a few things when checking performance: 

1. Look into a specific `#[bench]` in the file of the function I'm optimizing (or write one if the thing I'm optimizing isn't specifically benched). For instance, there is now a `bench_server_transaction_encode` in the `h1.rs` file, when I was writing the `date.rs` stuff.
2. I run `cargo bench --bench server` (or `--bench end_to_end` if the stuff I touched would affect the `Client` also).
3. I run `cargo run --release --example hello`, and then run [wrk](https://github.com/wg/wrk) to see if requests per second changed.

I compare the results of `master` vs my current work to determine if it was useful.